src/backscratcher/veFXSVault.sol

```diff
@@ -0,0 +1,546 @@
+                    ? checkpoints[dstRep][dstRepNum - 1].votes
+                    : 0;
+                uint256 dstRepNew = dstRepOld.add(amount);
+                _writeCheckpoint(dstRep, dstRepNum, dstRepOld, dstRepNew);
+            }
+        }
+    }
+
+    function _writeCheckpoint(
+        address delegatee,
+        uint32 nCheckpoints,
+        uint256 oldVotes,
+        uint256 newVotes
+    ) internal {
+        uint32 blockNumber = safe32(block.number, "_writeCheckpoint: 32 bits");
+
+        if (
+            nCheckpoints > 0 &&
+            checkpoints[delegatee][nCheckpoints - 1].fromBlock == blockNumber
+        ) {
+            checkpoints[delegatee][nCheckpoints - 1].votes = newVotes;
+        } else {
+            checkpoints[delegatee][nCheckpoints] = Checkpoint(
+                blockNumber,
+                newVotes
+            );
+            numCheckpoints[delegatee] = nCheckpoints + 1;
+        }
+
+        emit DelegateVotesChanged(delegatee, oldVotes, newVotes);
+    }
+
+    function safe32(uint256 n, string memory errorMessage)
+        internal
+        pure
+        returns (uint32)
+    {
+        require(n < 2**32, errorMessage);
+        return uint32(n);
+    }
+
+    /// @notice The standard EIP-20 transfer event
+    event Transfer(address indexed from, address indexed to, uint256 amount);
+
+    /// @notice The standard EIP-20 approval event
+    event Approval(
+        address indexed owner,
+        address indexed spender,
+        uint256 amount
+    );
+
+    /// @notice governance address for the governance contract
+    address public governance;
+    address public pendingGovernance;
+
+    IERC20 public constant FXS =
+        IERC20(0x853d955aCEf822Db058eb8505911ED77F175b99e);
+    address public constant LOCK =
+        address(0xF147b8125d2ef93FB6965Db97D6746952a133934);
+    address public proxy = address(0x7A1848e7847F3f5FfB4d8e63BdB9569db535A4f0);
+    address public feeDistribution;
+
+    IERC20 public constant rewards =
+        IERC20(0x853d955aCEf822Db058eb8505911ED77F175b99e);
+
+    uint256 public index = 0;
+    uint256 public bal = 0;
+
+    mapping(address => uint256) public supplyIndex;
+
+    function update() external {
+        _update();
+    }
+
+    function _update() internal {
+        if (totalSupply > 0) {
+            _claim();
+            uint256 _bal = rewards.balanceOf(address(this));
+            if (_bal > bal) {
+                uint256 _diff = _bal.sub(bal);
+                if (_diff > 0) {
+                    uint256 _ratio = _diff.mul(1e18).div(totalSupply);
+                    if (_ratio > 0) {
+                        index = index.add(_ratio);
+                        bal = _bal;
+                    }
+                }
+            }
+        }
+    }
+
+    function _claim() internal {
+        if (feeDistribution != address(0x0)) {
+            FeeDistribution(feeDistribution).claim(address(this));
```

Comment:
The rewards claiming interface is different from Curve's (so `claim(address)` is not correct).
Frax's rewards contract is at `0xed2647Bbf875b2936AAF95a3F5bbc82819e3d3FE` and the code is as follows:
```solidity
    function getYield() external nonReentrant notYieldCollectionPaused checkpointUser(msg.sender) returns (uint256 yield0) {
        require(greylist[msg.sender] == false, "Address has been greylisted");

        yield0 = yields[msg.sender];
        if (yield0 > 0) {
            yields[msg.sender] = 0;
            TransferHelper.safeTransfer(
                emitted_token_address,
                msg.sender,
                yield0
            );
            emit YieldCollected(msg.sender, yield0, emitted_token_address);
        }
    }
```